listeners.go

```diff
@@ -129,6 +139,10 @@ func (na NetworkAddress) ListenAll(ctx context.Context, config net.ListenConfig)
 // it even if the previous program using it exited uncleanly; it will also be
 // unlinked upon a graceful exit (or when a new config does not use that socket).
 func (na NetworkAddress) Listen(ctx context.Context, portOffset uint, config net.ListenConfig) (any, error) {
+	return na.ListenWithSocket(ctx, portOffset, config, "")
+}
+
+func (na NetworkAddress) ListenWithSocket(ctx context.Context, portOffset uint, config net.ListenConfig, socket string) (any, error) {
```

Comment:
Godoc; describe the socket parameter. (Do this for all new/changed methods.)

Reply:
all the `socket` and `socketFile` have been documented